src/transport/raw/MessageHeader.h

```diff
@@ -42,38 +42,49 @@
 
 namespace chip {
 
+static constexpr size_t kMaxTagLen = 16;
+
 namespace detail {
 // Figure out the max size of a packet we can allocate, including all headers.
-static constexpr size_t kMaxIPPacketSizeBytes       = 1280;
+static constexpr size_t kMaxUdpIPPacketSizeBytes       = 1280;
+static constexpr size_t kMaxTcpIPPacketSizeBytes       = 12800;
 static constexpr size_t kMaxUDPAndIPHeaderSizeBytes = 48;
 
-static_assert(kMaxIPPacketSizeBytes >= kMaxUDPAndIPHeaderSizeBytes + CHIP_SYSTEM_HEADER_RESERVE_SIZE,
+static_assert(kMaxUdpIPPacketSizeBytes >= kMaxUDPAndIPHeaderSizeBytes + CHIP_SYSTEM_HEADER_RESERVE_SIZE,
               "Matter headers and IP headers must fit in an MTU.");
 
 // Max space we have for our Application Payload and MIC, per spec.
 static constexpr size_t kMaxPerSpecApplicationPayloadAndMICSizeBytes =
```

Comment:
This is only the max for UDP.  Naming needs to be adjusted to make this clearer.